proto/catalog.proto

```diff
@@ -123,6 +124,8 @@ message Source {
   // Cluster version (tracked by git commit) when initialized/created
   optional string initialized_at_cluster_version = 17;
   optional string created_at_cluster_version = 18;
+  // Handle the source relies on any sceret. The key is the propertity name and the value is the secret id.
+  map<string, secret.SecretRef> secret_refs = 19;
```

Comment:
Why adding a new field in source proto? Isn't already exist in `StreamSourceInfo`?

Reply:
`StreamSourceInfo` for format-encode options, `PbSource` for connector related once. I agree that maybe too verbose in source but it is good to have a clear dep relation. d

Reply:
Please add comments

Reply:
may go with `format_encode_secret_refs` and `connector_secret_refs`

Reply:
Should `with_properties` in Source and `properties` both start with connector?

Reply:
I'll just add some comment for now. Let's do all the renaming in a separate PR. TBH I don't think it's particularly necessary. That fact that they are in `StreamSourceInfo` and `SinkFormatDesc` already tells they are for format/encode